File: backend/src/domain/entities/Produto.js
export class Produto {
  constructor({
    id,
    nome,
    descricao,
    preco,
    estoque,
    origem,
    altitudeM,
    processo,
    torra,
    imagemUrl,
    ativo = true
  }) {
    this.id = id;
    this.nome = nome;
    this.descricao = descricao;
    this.preco = preco;
    this.estoque = estoque;
    this.origem = origem;
    this.altitudeM = altitudeM;
    this.processo = processo;
    this.torra = torra;
    this.imagemUrl = imagemUrl;
    this.ativo = ativo;
  }

  reduzirEstoque(quantidade) {
    if (this.estoque < quantidade) {
      throw new Error('Estoque insuficiente');
    }
    this.estoque -= quantidade;
  }
}

File: backend-java/src/main/java/com/paralelo14/domain/entity/Produto.java
package com.paralelo14.domain.entity;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "produtos")
public class Produto extends AbstractStringIdEntity {

    @Column(nullable = false)
    private String nome;

    @Column(columnDefinition = "TEXT")
    private String descricao;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal preco;

    @Column(nullable = false)
    private Integer estoque;

    private String origem;
    @Column(name = "altitudeM")
    private Integer altitudeM;
    private String processo;
    private String torra;

    @Column(name = "imagemUrl", columnDefinition = "TEXT")
    private String imagemUrl;

    @Column(nullable = false)
    private Boolean ativo = true;

    @JsonIgnore
    @OneToMany(mappedBy = "produto")
    private List<ItemPedido> itensPedido = new ArrayList<>();

    @JsonIgnore
    @OneToMany(mappedBy = "produto", cascade = CascadeType.ALL)
    private List<Avaliacao> avaliacoes = new ArrayList<>();
}
